projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/DataJobNotification.java

```diff
@@ -81,11 +87,19 @@ public void notifyJobDeployError(JobConfig jobConfig, String errorName, String e
               ownerName,
               ownerEmail,
               ccEmails);
-      notification.send(notificationContent);
+      sendNotification(notificationContent);
     } catch (AddressException e) {
       log.warn("Could not send notification due to bad email format", e);
     } catch (MessagingException e) {
       log.warn("Could not send notification due to message error", e);
     }
   }
+
+  private void sendNotification(NotificationContent notificationContent) throws MessagingException {
+    if (configuration.isAuthEnabled()) {
+      authenticatedEmailNotification.send(notificationContent);
+    } else {
+      notification.send(notificationContent);
+    }
+  }
```

Comment:
AuthenticatedEmailNotification is an EmailNotification but with added authentication. 

The DataJobNotification class doesn't have to know whether the notification needs to be authenticated or not. It will only depend on an abstraction (EmailNotification interface send(... ), not on any concrete implementation.   

This aligns well with the principles of good object-oriented design specifically -  the dependency inversion principle (depend on abstractions - the EmailNotification interface, not on any concrete implementation) and encapsulation (hide implementation details).